code/game/gamemodes/vampire/vampire.dm

```diff
@@ -227,6 +227,7 @@ You are weak to holy things and starlight. Don't go into space and avoid the Cha
 	if(istype(spell, /obj/effect/proc_holder/spell))
 		owner.mind.AddSpell(spell)
 	powers += spell
+	owner.update_sight()
```

Comment:
somehow feel like this should be specifically for the /datum/vampire_passive/vision path